static/js/controllers/edit-user.js

```diff
@@ -133,12 +143,18 @@
         if (!updatingSelf) {
           // users don't have permission to update their own security settings
           settings.roles = getRoles($scope.editUserModel.type, true);
-          settings.facility_id = $('#edit-user-profile [name=facility]').val();
-          settings.contact_id = $('#edit-user-profile [name=contact]').val();
+          settings.facility_id = $scope.editUserModel.facility_id;
+          settings.contact_id = $scope.editUserModel.contact_id;
         }
         return settings;
       };
 
+      var computeFields = function() {
```

Comment:
Select2 fields don't work happily with the model without extra work. 
Filed : Non-angular select 2 fields : #3444